### ISSUE TITLE
feat: alternatieve ML-modellen onderzoeken voor cumulatief spoor

### ISSUE THREAD

**Issue by Muhammet369** [ed2c/studentprognose#147]
## Context

Het cumulatieve spoor (`-d cumulative`) gebruikt een vaste twee-staps pipeline:

1. **SARIMA** (`src/studentprognose/models/sarima.py:predict_with_sarima_cumulative`) —
   extrapoleert de cumulatieve gewogen-vooraanmelderscurve tot week 38.
2. **XGBoost Regressor** (`src/studentprognose/models/xgboost_regressor.py`) — voorspelt
   het eindaantal studenten op basis van de SARIMA-curve + engineered features
   (lags, acceleration, exclusivity ratio).

Net als bij het individuele spoor is deze combinatie historisch gegroeid en niet
systematisch vergeleken met alternatieven.

## Probleem

- **SARIMA** met vaste ordes is rigide en faalt stil bij convergentieproblemen
  (warnings worden onderdrukt in `cumulative.py:9`).
- **XGBoost Regressor** met alleen `learning_rate=0.25` is minimaal geconfigureerd.
  Andere regressors die beter omgaan met kleine datasets (weinig trainingsjaren)
  zijn niet onderzocht.
- De engineered features (`Gewogen_t-2`, `Gewogen_t-5`, `acceleration`,
  `exclusivity_ratio`) zijn handmatig gekozen — er is geen feature-selectie of
  -validatie gedaan.
- Bij numerus fixus-opleidingen wordt een apart XGBoost-model per opleiding
  getraind, maar met zeer weinig trainingsdata (< 10 rijen). Dit is fragiel.

### Stap 1: Tijdreeksextrapolatie (vervanging/aanvulling SARIMA)

| Model | Voordeel | Nadeel |
|---|---|---|
| **ETS** | Stabieler bij korte reeksen, minder parameters | Geen exogene variabelen |
| **Prophet** | Automatische seizoenaliteit, uncertainty intervals | Zware dependency |
| **VAR (Vector AR)** | Modelleert meerdere opleidingen tegelijk, vangt correlaties | Complexer, meer data nodig |
| **Theta-methode** | Zeer simpel, competitief bij korte reeksen (M3-competitie) | Geen seizoenaliteit |

### Stap 2: Regressie (vervanging/aanvulling XGBoost Regressor)

| Model | Voordeel | Nadeel |
|---|---|---|
| **LightGBM Regressor** | Sneller, betere categorische support | Extra dependency |
| **Ridge/Lasso Regression** | Stabiel bij weinig data, interpreteerbaar | Lineair, mist interacties |
| **Elastic Net** | Combineert L1+L2 regularisatie, goed bij multicollineariteit | Lineair |
| **Random Forest Regressor** | Robuust bij kleine datasets, minder overfitting | Minder nauwkeurig dan boosting bij genoeg data |

### Stap 3: End-to-end alternatieven

| Model | Voordeel | Nadeel |
|---|---|---|
| **Direct multi-step regression** | Voorspelt week 38 direct zonder SARIMA-tussenstap | Verliest tijdreeksinformatie |
| **Gradient boosting op ruwe tijdreeks** | Eén model i.p.v. twee stappen | Moeilijker te interpreteren |

### AutoTS als benchmark-tool

[AutoTS](https://github.com/winedarksea/AutoTS) kan ingezet worden om automatisch tientallen tijdreeksmodellen (ARIMA, ETS, Theta, Prophet, etc.) te testen en ranken op de cumulatieve vooraanmelderscurves. Dit is bijzonder waardevol voor het cumulatieve spoor omdat de tijdreekscomponent hier de eerste stap is — een beter tijdreeksmodel verbetert direct de input voor de XGBoost Regressor.

| Voordeel | Nadeel |
|---|---|
| Test automatisch veel modellen in één run | Zware dependency met veel transitive deps |
| Ingebouwde cross-validatie en metric-vergelijking | Bij korte reeksen (5-8 jaar) mogelijk instabiele modelselectie |
| Ondersteunt ensembles van tijdreeksmodellen | Vervangt alleen het tijdreeksdeel, niet de XGBoost-regressie |
| Kan per opleiding×herkomst het beste model selecteren | Black-box: minder controle over waarom een model gekozen wordt |

**Aanbeveling:** gebruik AutoTS in Fase 1 als exploratietool om de meest belovende modelfamilies te identificeren, en implementeer de winnaar(s) vervolgens handmatig in de pipeline voor volledige controle. Meet daarbij apart de SARIMA-curve MAPE (tussenresultaat) en de eindvoorspelling MAPE (na XGBoost) om te zien waar de winst zit.

### Fase 1: Benchmark-framework

1. Zelfde framework als het individuele spoor (zie #145): **TimeSeriesSplit**
   met train t/m jaar N-1, test op jaar N.
2. Meet: **MAPE**, **MAE**, **RMSE**, **trainingstijd**.
3. Specifiek voor cumulative: meet ook de **SARIMA-curve MAPE** (tussenresultaat)
   apart van de **eindvoorspelling MAPE** (na XGBoost).
4. **Draai AutoTS** op de cumulatieve gewogen-vooraanmelderscurves om snel de meest belovende tijdreeksmodellen te identificeren voordat handmatige experimenten starten.

### Fase 2: Modelexperimenten

5. Train de top-modellen uit de AutoTS-scan als tijdreeks-alternatief naast SARIMA.
6. Train Ridge Regression en Random Forest als regressor-alternatieven naast XGBoost.
7. Test of een direct multi-step model (zonder SARIMA-tussenstap) competitief is.
8. Onderzoek specifiek de numerus fixus-case: is een apart model per opleiding
   beter dan een gedeeld model met een numerus fixus-feature?

### Fase 3: Integratie

9. Beste model(len) configureerbaar via `configuration.json`:
   ```json
   "model_config": {
       "cumulative_timeseries": "sarima",
       "cumulative_regressor": "xgboost"
   }
   ```

## Acceptatiecriteria

- [ ] Benchmark-framework (gedeeld met individueel spoor) is beschikbaar
- [ ] AutoTS-scan is gedraaid en resultaten zijn gedocumenteerd
- [ ] Minimaal 2 alternatieve regressors zijn vergeleken met XGBoost
- [ ] Minimaal 1 alternatief tijdreeksmodel is vergeleken met SARIMA
- [ ] Numerus fixus-aanpak is geëvalueerd (apart vs. gedeeld model)
- [ ] Resultaten zijn gedocumenteerd (MAPE/MAE per model per groep)
- [ ] Beste model is configureerbaar via `configuration.json`

## Referenties

- Huidige pipeline: `CumulativeStrategy` in `src/studentprognose/strategies/cumulative.py`
- SARIMA: `src/studentprognose/models/sarima.py`
- XGBoost Regressor: `src/studentprognose/models/xgboost_regressor.py`
- Engineered features: `cumulative.py:20-23` (`ENGINEERED_FEATURE_COLS`)
- AutoTS: https://github.com/winedarksea/AutoTS
- Gerelateerd: #111 (hyperparameter tuning), #145 (individueel spoor)

**Comment by Muhammet369** [comment]
## Benchmark: alternatieve ML-modellen voor cumulatief spoor

Benchmark gedraaid op eigen data (week 12, leave-last-year-out cross-validatie, 324 tijdreeks-evaluaties, 10 regressie-evaluaties).

### Stap 1: Tijdreeksmodellen (curve-extrapolatie)

| # | Model | MAPE | MAE | RMSE | Tijd/fit | Convergentie | Status |
|---|-------|------|-----|------|----------|--------------|--------|
| 🥇 | **CES** | **0.0387** | **2.50** | **2.90** | 0.109s | 100% | ⚠️ nog niet in pipeline |
| 🥈 | Theta | 0.0401 | 2.60 | 3.01 | 0.024s | 100% | ✅ beschikbaar |
| 🥉 | MSTL | 0.0402 | 2.59 | 2.97 | 0.023s | 100% | ⚠️ nog niet in pipeline |
| 4 | **SARIMA** | 0.0466 | 2.96 | 3.37 | 1.638s | 100% | ✅ huidige default |
| 5 | ETS | 0.0613 | 3.94 | 4.27 | 0.080s | 100% | ✅ beschikbaar |
| 6 | SeasonalNaive | 0.0673 | 4.37 | 4.70 | 0.000s | 100% | ⚠️ nog niet in pipeline |
| 7 | AutoARIMA | — | — | — | — | — | ⏱️ te traag (>30 min) |

### Stap 2: Regressiemodellen (vooraanmelders → inschrijvingen)

| # | Model | MAPE | MAE | RMSE | Tijd/fit | Train N | Status |
|---|-------|------|-----|------|----------|---------|--------|
| 🥇 | **CatBoost** | **0.1436** | 7.09 | 13.09 | 0.354s | 135 | ⚠️ nog niet in pipeline |
| 🥈 | Random Forest | 0.1439 | **6.86** | **11.52** | 0.190s | 135 | ✅ beschikbaar |
| 3 | **XGBoost** | 0.1555 | 7.79 | 13.28 | 2.693s | 135 | ✅ huidige default |
| 4 | Ridge | 0.2479 | 7.89 | 11.67 | 0.001s | 135 | ✅ beschikbaar |
| 5 | LightGBM | 0.2848 | 13.58 | 22.72 | 1.799s | 135 | ⚠️ slecht bij kleine N |

### AutoTS verkenning (eenmalig, 6 grootste groepen)

| # | Model | Score | MAE |
|---|-------|-------|-----|
| 1 | DatepartRegression (DecisionTree + wavelets) | 15.48 | 2.47 |
| 2 | BallTreeMultivariateMotif | 15.99 | 3.64 |
| 3 | SectionalMotif | 17.51 | 3.53 |

AutoTS selecteert motif-gebaseerde en wavelet-methoden als winnaars — interessant als toekomstige onderzoeksrichting.

### Conclusie

| | Huidige default | Beste alternatief | Verschil |
|---|----------------|-------------------|----------|
| Tijdreeks | SARIMA (MAPE 0.047) | CES (MAPE 0.039) | **−17% fout, 15× sneller** |
| Regressie | XGBoost (MAPE 0.156) | CatBoost (MAPE 0.144) | **−8% fout, 8× sneller** |

### Aanbeveling

- **CES + Random Forest** als we geen extra dependency willen (Random Forest ≈ CatBoost, geen `catboost` package nodig)
- **CES + CatBoost** als extra dependency geen bezwaar is
- CES en MSTL nog toevoegen als configureerbare opties in de pipeline

**Comment by Muhammet369** [comment]
## Benchmark resultaten: cumulatief + individueel spoor

Implementatie van het benchmark-framework voor beide sporen. Het individuele spoor (#145) is ook in deze branch meegenomen.

### Wat is er gebouwd
- HTML benchmark rapporten per spoor: `benchmark_cumulative.html` (`-d c`) en `benchmark_individual.html` (`-d i`)
- Interactieve Chart.js visualisaties: bar charts, line charts per testjaar, en ROC curves (individueel)
- **Cumulatief**: Stap 1 vergelijkt 3 tijdreeksmodellen (SARIMA, ETS, Theta), Stap 2 vergelijkt 5 regressors
- **Individueel**: Stap 1 vergelijkt 5 classifiers met interactieve ROC curves (mean + confidence band, per-jaar weergave, model toggle)
- TimeSeriesSplit evaluatie: train t/m jaar N-1, test op jaar N

### Hoe draai je de benchmark?

\`\`\`bash
# Cumulatief spoor (tijdreeks + regressie)
uv run studentprognose benchmark -d c -w 16 -y 2024

# Individueel spoor (classificatie + ROC curves)
uv run studentprognose benchmark -d i -w 16 -y 2024
\`\`\`

Rapporten worden automatisch gegenereerd in \`data/output/benchmark/\`.

### Resultaten (demo-data)

**Cumulatief — Stap 1: Tijdreeks**

| Model | MAPE | MAE | RMSE | Tijd (s) | Conv. |
|---|---|---|---|---|---|
| Theta | 0.0334 | 2.4 | 2.7 | 0.017 | 100% |
| SARIMA | 0.0405 | 2.8 | 3.2 | 1.580 | 100% |
| ETS | 0.0578 | 4.1 | 4.4 | 0.067 | 100% |

**Cumulatief — Stap 2: Regressie**

| Model | MAPE | MAE | RMSE | Tijd (s) |
|---|---|---|---|---|
| Gradient Boosting | 0.1342 | 6.6 | 11.2 | 0.083 |
| Extra Trees | 0.1360 | 6.4 | 10.7 | 0.095 |
| Random Forest | 0.1445 | 6.8 | 11.5 | 0.139 |
| XGBoost | 0.1603 | 7.8 | 13.1 | 0.052 |
| Ridge | 0.2486 | 7.9 | 11.7 | 0.000 |

**Individueel — Stap 1: Classificatie**

| Model | Accuracy | AUC-ROC | F1 | Agg MAE | Tijd (s) |
|---|---|---|---|---|---|
| Random Forest | 0.6766 | 0.5016 | 0.8068 | 4.0 | 4.8 |
| Gradient Boosting | 0.6750 | 0.5014 | 0.8053 | 2.1 | 2.9 |
| XGBoost | 0.6649 | 0.4993 | 0.7969 | 2.2 | 1.4 |
| Logistic Regression | 0.6516 | 0.4985 | 0.7813 | 2.4 | 28.9 |
| Extra Trees | 0.6728 | 0.4971 | 0.8031 | 3.8 | 6.7 |

### ROC curves zijn diagonaal — verwacht met demo-data

Alle classifiers scoren AUC ~0.50 (random-niveau). Dit is geen bug: de synthetische demo-data kent inschrijfstatus willekeurig toe (\`rng.shuffle\`), onafhankelijk van features. Met echte institutionele data verwachten we dat de ROC curves naar linksboven buigen.

### Status

Hiermee zijn #145 en #147 Fase 1 (benchmark-framework) afgerond. Fase 2 (AutoTS-scan) en Fase 3 (model configureerbaar via \`configuration.json\`) slaan we bewust over — overengineering voor de huidige scope.
